src/Symfony/Component/Workflow/EventListener/GuardListener.php

```diff
@@ -28,15 +29,17 @@ class GuardListener
     private $authenticationChecker;
     private $trustResolver;
     private $roleHierarchy;
+    private $validator;
 
-    public function __construct($configuration, ExpressionLanguage $expressionLanguage, TokenStorageInterface $tokenStorage, AuthorizationCheckerInterface $authenticationChecker, AuthenticationTrustResolverInterface $trustResolver, RoleHierarchyInterface $roleHierarchy = null)
+    public function __construct($configuration, ExpressionLanguage $expressionLanguage, TokenStorageInterface $tokenStorage, AuthorizationCheckerInterface $authenticationChecker, AuthenticationTrustResolverInterface $trustResolver, RoleHierarchyInterface $roleHierarchy = null, ValidatorInterface $validator = null)
```

Comment:
I don't know how many types of conditions can bloc a transistion but yes we can imagine a lot of thing.

Then I could split `GuardListener` into `SecurityGuardListener` (the same as GuardListener before my PR) and a `ValidatorGuardListener` with the new behavior.

In FrameworkExtension, I need to create a new definition for `ValidatorGuardListener`.
For the definition id are you ok with :

```
sprintf('%s.listener.guard.security', $workflowId)
and
sprintf('%s.listener.guard.validator', $workflowId)
```

And I plug it on the same event ?

```
sprintf('workflow.%s.guard.%s', $name, $transitionName)** 
```